controllers/rollingupgrade_controller.go

```diff
@@ -262,8 +260,8 @@ func (r *RollingUpgradeReconciler) WaitForDesiredInstances(ruObj *upgrademgrv1al
 			return err
 		}
 
-		inServiceCount := getInServiceCount(r.Asg.Instances)
-		if inServiceCount == aws.Int64Value(r.Asg.DesiredCapacity) {
+		inServiceCount := getInServiceCount(r.Asgs[ruObj.Name].Instances)
```

Comment:
Potentially, if we make sure `ruObjNameToASG` gets updated by `populateAsg`, we can then load, cast and use it that way.

Example
```
	value, ok := r.ruObjNameToASG.Load(ruObj.Name)
	if !ok {
		msg := "Failed to find rollingUpgrade name in map."
		log.Printf(msg)
		return 0, errors.New(msg)
	}

	asg := value.(*autoscaling.Group)
```

and then
```
inServiceCount := getInServiceCount(asg.Instances)
```